### DISCUSSION TITLE
Win32com shell: list "Dieser PC" with Apple Iphone - Issue with pywebview

### DISCUSSION THREAD

**Discussion by baschni** [mhammond/pywin32#2534] [General]
Hello,

using pywin32 together with pywebview I noticed unexpected behaviour. I would like to use the shell to list everything in the windows folder "Dieser PC"/"This computer".

I made a minimal demonstration example:

```
from win32com.shell import shell, shellcon
from webview import create_window, start, windows

class Api():
    def listFoldersDieserPC(self):

        folder = shell.SHGetDesktopFolder()
        for item in folder:
            print("loop: ", folder.GetDisplayNameOf(item, shellcon.SHGDN_NORMAL))
            if folder.GetDisplayNameOf(item, shellcon.SHGDN_NORMAL) == "Dieser PC":
                new_folder = folder.BindToObject(item, None, shell.IID_IShellFolder)
                break

        # list folders in Dieser PC/This Computer
        for item in new_folder:
            print("loop: ", new_folder.GetDisplayNameOf(item, shellcon.SHGDN_NORMAL))


api = Api()
window = create_window('Test', 'index.html', js_api=api, min_size=(600, 450))
api.listFoldersDieserPC()
print("now from the browser")
start()

```

With index.html:
`<button value="asdf" onClick="window.pywebview.api.listFoldersDieserPC()">List Folders</button>`

The first time I call the function api.listFoldersDieserPC directly from python, the second time through the button in the pywebview browser window.

This is the output:

> loop:  Dieser PC
> loop:  Downloads
> loop:  Bilder
> loop:  Musik
> loop:  Desktop
> loop:  Dokumente
> loop:  Videos
> loop:  Apple iPhone
> loop:  Local Disk (C:)
> now from the browser
> loop:  Dieser PC
> loop:  Downloads
> loop:  Bilder
> loop:  Musik
> loop:  Desktop
> loop:  Dokumente
> loop:  Videos
> loop:  Local Disk (C:)
> 
> Process finished with exit code 0

It is the same, except that the folder for Apple iPhone is missing on the second go.

Why is the shell behaving differently here? What can I change to make it show the "Apple iPhone" entry also on the function call from the pywebview browser?

Kind regards
Bastian

**Comment by mhammond** [comment]
[This sounds like a support request](https://github.com/mhammond/pywin32/blob/main/README.md#support)
